src/actions/actionsFactory.ts

```diff
@@ -0,0 +1,39 @@
+import { Uri } from "vscode";
+import { IAction, ICommandAction, ActionType, IFileAction, ActionJsonKey } from './interfaces';
+import { CommandAction, FileAction } from './impl';
+import _ = require("lodash");
+
+export class ActionsFactory {
+    public static createAction(jsonAction: any) : IAction {
+        const actionType = jsonAction[ActionJsonKey.ActionType];
+        if (_.isEmpty(actionType)) {
+            throw new Error(`${ActionJsonKey.ActionType} is missing`);
+        }
+        switch (actionType) {
```

Comment:
only two action types? (case ActionType.Command , case ActionType.File)

Reply:
Yes, 2 actions for now. We will add in case of requirement. Currently it is not necessary. Ido is required it in his way.